lib/data_magic/config.rb

```diff
@@ -302,36 +317,38 @@ def index_needs_update?(index_name=nil)
     #   s3://bucket_name
     #   a local path like: './data'
     def read_path(path)
+      result = nil
       uri = URI(path)
       scheme = uri.scheme
       case scheme
         when nil
           begin
```

Comment:
This method is a bit long - how about breaking the two cases out into `read_path_local` and `read_path_s3`?